src/highdicom/sr/content.py

```diff
@@ -103,6 +163,35 @@ def __init__(
             relationship_type=RelationshipTypeValues.INFERRED_FROM
         )
 
+    @classmethod
+    def from_source_image(
+            cls,
+            image_dataset: Dataset,
```

Comment:
Do you have strong feelings about using `image_dataset` instead of `image`? I generally avoid including the type in the name. We have a type annotation and a docstring to inform the user.

Reply:
Just that 'image' is very vague and could be a numpy pixel array or something. On balance though it is probably better. Have changed to `image` in this class and all other `SourceXForX` classes, and additionally changed the `dataset` parameter of `ReferencedSegment.from_seg_dataset()` to `segmentation` to match. Changes in 4cc5f0e

On a related note, should we change `from_seg_dataset` to simply `from_seg` or `from_segmentation` to better match `from_source_image`?